src/pages/index.js

```diff
@@ -37,61 +60,106 @@ profileEditButton.addEventListener("click", () => {
+    editProfilePopup.close();
+    editFormValidator.toggleButtonState();
+  });
+});
+editProfilePopup.setEventListeners();
+
+addNewCardButton.addEventListener("click", () => {
+  addCardPopup.open();
+});
+
 const renderCard = (cardData) => {
   const cardElement = new Card(
     cardData,
     selectors.cardTemplate,
-    handleImageClick
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
   );
   cardSection.addItem(cardElement.getView());
 };
 
-function handleAddCardSubmit(data) {
-  renderCard(data);
-  addCardPopup.close();
-  addFormValidator.toggleButtonState();
-}
-
-function handleEditProfileSubmit(data) {
-  userInfo.setUserInfo(data);
-  editProfilePopup.close();
-  editFormValidator.toggleButtonState();
-}
+const addCardPopup = new PopupWithForm(addCardModal, (data) => {
+  addCardPopup.saving(true);
+  api.addCard(data).then((data) => {
+    addCardPopup.saving(false);
+    renderCard(data);
+    addCardPopup.close();
+    addFormValidator.toggleButtonState();
+  });
+});
+addCardPopup.setEventListeners();
 
 function handleImageClick(name, link) {
   previewImagePopup.open(name, link);
 }
 
-const cardSection = new Section(
-  {
-    renderer: renderCard,
-    items: initialCards,
-  },
-  selectors.cardSection
-);
+const previewImagePopup = new PopupWithImage(imageModal, handleImageClick);
+previewImagePopup.setEventListeners();
 
-cardSection.renderItems();
+function handleLikeClick(card) {
+  if (card.isLiked) {
+    api.removeLike(card._id).then(() => {
+      card.handleLikeButton();
+    });
+  } else {
+    api.likeCard(card._id).then(() => {
+      card.handleLikeButton();
+    });
+  }
+}
 
-const addCardPopup = new PopupWithForm(addCardModal, handleAddCardSubmit);
-addCardPopup.setEventListeners();
+const deleteCardPopup = new PopupWithConfirm(deleteCardModal);
+deleteCardPopup.setEventListeners();
 
-const editProfilePopup = new PopupWithForm(
-  profileEditModal,
-  handleEditProfileSubmit
-);
-editProfilePopup.setEventListeners();
+function handleDeleteClick(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.deleting(true);
+    api.deleteCard(card._id).then(() => {
+      deleteCardPopup.deleting(false);
+      card.removeCard();
+      deleteCardPopup.close();
+    });
+  });
+}
 
-const previewImagePopup = new PopupWithImage(imageModal, handleImageClick);
-previewImagePopup.setEventListeners();
+avatarEditButton.addEventListener("click", () => {
+  editAvatarPopup.open();
+});
 
-const userInfo = new UserInfo(profileTitle, profileDescription);
+const editAvatarPopup = new PopupWithForm(avatarEditModal, (data) => {
+  editAvatarPopup.saving(true);
+  api.setUserAvatar(data).then((data) => {
+    editAvatarPopup.saving(false);
+    userInfo.setAvatar(data.avatar);
+    editAvatarPopup.close();
+    avatarFormValidator.toggleButtonState();
+  });
+});
+editAvatarPopup.setEventListeners();
 
 const editFormValidator = new FormValidator(
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```